### DISCUSSION TITLE
Request to add parameter for new MagickImage(color, width, height) overload

### DISCUSSION THREAD

**Discussion by ghost** [dlemstra/Magick.NET#1045] [Help]
It took a bit of experimentation to fix a "no encode delegate for this image format 'XC'" error I was receiving in .Net 5 w/ Magick.Net Q8-x64 8.3.2 using new MagickImage(color, width, height) -- I had to add { Format = MagickFormat.whatever } which I don't remember needing in the past. If the format is required for object creation to work, it would be nice to have it in the parameters for the overload.

**Comment by dlemstra** [comment]
Do you have a code sample that can be used to reproduce this issue?

**Comment by ghost** [comment]
`//Throws exception`
`using (MagickImage mi = new(MagickColors.Black, 1280, 600))`
`//Works`
`using (MagickImage mi = new(MagickColors.Black, 1280, 600) { Format = MagickFormat.Png })`
`//Desired`
`using (MagickImage mi = new(MagickColors.Black, 1280, 600, MagickFormat.Png))`
`//or`
`using (MagickImage mi = new(MagickFormat.Png, MagickColors.Black, 1280, 600))`

**Comment by dlemstra** [comment]
Thanks for explaining this but this is more an output format issue than an input format issue. You are probably getting the exception when you try to write the image. Most `Write` overloads allow specifying the `MagickFormat` so you could do something like this: `mi.Write(output, MagickFormat.Png)`.
